### DISCUSSION TITLE
Event handler when currently selected instruction changes

### DISCUSSION THREAD

**Discussion by thomasfaingnaert** [Vector35/binaryninja-api#2153] [API] [answered]
Is there a way for a Python plugin to register an event handler that is called whenever the user selects a different instruction in e.g. the disassembly view?

**Comment by plafosse** [comment]
Currently the only way to get this is to implement a DockWidget. We plan to add this to the UI api's as well.

**Comment by CouleeApps** [comment]
Tracking issue: https://github.com/Vector35/binaryninja-api/issues/2166

**Comment by CouleeApps** [comment, marked as answer]
This is now supported as of dev 2.2.2557. See this example: https://github.com/Vector35/binaryninja-api/blob/432a8cdf64052db5fef7df792e2a5a51734d27af/python/examples/ui_notifications.py#L74

**Reply by thomasfaingnaert** [reply]
Great, thanks for the quick fix!
